hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -360,19 +349,40 @@ private long calculateGasUsedByTX(final long txGasLimit, final MessageFrame init
 		return gasUsedByTransaction;
 	}
 
-	protected long gasPriceTinyBarsGiven(final Instant consensusTime, boolean isEthTxn) {
+	public long calculateGasUsedByTokenCallTX(final long txGasLimit, final long remainingGas) {
+		long gasUsedByTransaction = txGasLimit - remainingGas;
```

Comment:
We can simulate the remaining gas calculation by the following logic:

The `remainingGas` value inside the EVM is assigned with the `initialGas` value, which we can calculate. Since we don't spin up the EVM it doesn't consume any gas, so we can just use the `initialGas` as the `remainingGas`.

The `initialGas` is:

`final var gasAvailable = gasLimit - intrinsicGas;`

where:

`final long intrinsicGas = gasCalculator.transactionIntrinsicGasCost(Bytes.EMPTY, contractCreation);`

and gasLimit is the `amount` from contract_call.proto in the case of contract call.